lib/partial/transmuxer.js

```diff
@@ -134,27 +134,35 @@ var tsPipeline = function(options) {
     });
   });
 
-  /* TODO caption support
-   * Caption support may be trickier than for full segment appends, as the captions can
-   * be split and processed in a much more granular fashion.
-  pipeline.captionStream.on('data', function(data) {
-    pendingCaptions.push(data);
-  });
+  pipeline.captionStream.on('data', function(caption) {
+    var timelineStartPts;
+    var startTime;
+    var endTime;
+
+    if (pipeline.tracks.video) {
+      timelineStartPts = pipeline.tracks.video.timelineStartInfo.pts || 0;
+    } else {
+      // TODO: how do we handle the case where captions are returned before
```

Comment:
Is that a possibility?

Reply:
I think the captions spec implies that video packets are specified before caption data, but I think it's still technically possible(though unlikely) that we could have caption data before video data.

Reply:
Going to leave this as a comment as it's an unlikely scenario(though we do work around it in mp4/transmuxer.js with the CoalesceStream)